### PR TITLE
INT B-21429 and B-21431 combo for populating shipment market code

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

We want to be able to track if a shipment is a domestic/international shipment, but we can only do that by checking the addresses under the shipment. If both addresses are CONUS then the `market_code` should be `d`, else it's gonna be `i`.

This does contain code needed for [B-21430](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21430) since the `isOconus` value is necessary to determine the shipment's market code.

This PR does the following:
- migration to populate `market_code` columns
- updates all Swagger definitions for shipments to send back `marketCode` to the UI
- updates all `model_to_payload.go` files to send back `marketCode`
- adds in logic in both shipment creation and update that determines if `marketCode` needs to change by checking the address values
- adds/updates relevant tests

### How to test

1. Fire up yer client & server
2. Check your migrations and the `mto_shipments` table - every value in `market_code` should say `d`
3. Create any shipment as a customer/office user with any CONUS addy
4. Throw in some console logs somewhere that you can see the data
5. You should see the `marketCode` value in the shipment object
6. Now... To test if a shipment changes to `i` you need to be able to enter an OCONUS state (in this case, Alaska)
7. Close down your server, in your `envrc` turn `FEATURE_FLAG_ENABLE_ALASKA` to `true`
8. Run `direnv allow` and fire up your server again
9. Create a shipment with an Alaska address
10. Check your logs, should now see `i` in the `marketCode` value on the shipment

## Screenshots
Screenshot 2024-10-10 at 1 29 48 PM

### CODE REVIEW

File: pkg/services/orchestrators/shipment/shipment_creator.go [hunk 1]
```diff
@@ -83,14 +82,13 @@ func (s *shipmentCreator) CreateShipment(appCtx appcontext.AppContext, shipment
 			mtoShipment.PPMShipment.ShipmentID = mtoShipment.ID
 			mtoShipment.PPMShipment.Shipment = *mtoShipment
 
-			_, err = s.ppmShipmentCreator.CreatePPMShipmentWithDefaultCheck(txnAppCtx, mtoShipment.PPMShipment)
-
+			_, err := s.ppmShipmentCreator.CreatePPMShipmentWithDefaultCheck(txnAppCtx, mtoShipment.PPMShipment)
```
Comment (TevinAdams) [inline pkg/services/orchestrators/shipment/shipment_creator.go:85]: Was this change intended or needed? Asking because thats the only change I see besides removing th eempty lines.

**Review by TevinAdams — APPROVED** [review]
Code reviewed and tested. ACs met. Was able to create CONUS and OCONUS shipment and see the marketCode value reflect the move type in the shipments object.

Screenshot 2024-10-11 at 7 52 17 AM
Screenshot 2024-10-11 at 7 52 03 AM

File: pkg/services/orchestrators/shipment/shipment_creator.go (hunk 1 above)
Comment (danieljordan-caci) [reply pkg/services/orchestrators/shipment/shipment_creator.go:85]: Ah, this was leftover from handling updating the shipment in this transaction, but I moved it over to the ppmShipment function instead - I'll change from `:=` to `=`

**Review by cameroncaci — CHANGES_REQUESTED** [review]
When the prime updates destination address and sets state to AK within 50 miles of zip code it does not adjust the market code from `d` -> `i` or `i` -> `d`
image

However it does work perfectly for office users updating the shipment

**Comment by danieljordan-caci** [comment]
> When the prime updates destination address and sets state to AK within 50 miles of zip code it does not adjust the market code from `d` -> `i` or `i` -> `d` image
> 
> However it does work perfectly for office users updating the shipment

Ah, nice catch. That must use a separate updater function. I'll take a look.

File: migrations/app/schema/20241008212243_populate_market_code_on_shipments_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-- Populate the new market_code column for shipments
+-- since we do not support OCONUS moves yet, these should all be "d"
+UPDATE mto_shipments
+SET market_code = 'd'
+WHERE market_code IS NULL;
+
+-- fixing typo from previous migration
+COMMENT ON COLUMN mto_shipments.market_code IS 'Market code indicator for the shipment. i for international and d for domestic.';
```
Comment (cameroncaci) [inline migrations/app/schema/20241008212243_populate_market_code_on_shipments_table.up.sql:8]: Even though this is such a simple query I worry it could timeout. Thoughts on bumping it?

Comment (danieljordan-caci) [reply migrations/app/schema/20241008212243_populate_market_code_on_shipments_table.up.sql:8]: Don't mind putting an increased timeout on it - won't hurt.

**Comment by danieljordan-caci** [comment]
> When the prime updates destination address and sets state to AK within 50 miles of zip code it does not adjust the market code from `d` -> `i` or `i` -> `d` image
> 
> However it does work perfectly for office users updating the shipment

This was addressed [in this commit](e4b112f0491f1a37d1d3cc0ad9741f9e9da0eca0)

File: migrations/app/schema/20241008212243_populate_market_code_on_shipments_table.up.sql (hunk 2 above)
Comment (danieljordan-caci) [reply migrations/app/schema/20241008212243_populate_market_code_on_shipments_table.up.sql:14]: Added timeout increase [here](af67917a724728da613c523ed86cda3fc5d34277)

File: pkg/services/address/address_creator.go [hunk 3]
```diff
@@ -42,15 +42,44 @@ func (f *addressCreator) CreateAddress(appCtx appcontext.AppContext, address *mo
...
 			return nil, err
 		}
+		transformedAddress.Country = &country
 		transformedAddress.CountryId = &country.ID
 	}
 
+	// use the data we have first, if it's not nil
+	if transformedAddress.Country != nil {
+		country := transformedAddress.Country
+		if country.Country != "US" || country.Country == "US" && transformedAddress.State == "AK" || country.Country == "US" && transformedAddress.State == "HI" {
```
Comment (traskowskycaci) [inline pkg/services/address/address_creator.go:59]: Do the country.Country == "US" && transformedAddress.State == "AK" and country.Country == "US" && transformedAddress.State == "HI" need to be in parentheses?

**Review by traskowskycaci — COMMENTED** [review]
Had one question, but testing looks good

image

AK:
image

File: swagger/support.yaml [hunk 4]
```diff
@@ -1579,6 +1579,16 @@ definitions:
       pointOfContact:
         type: string
         description: Email or id of a contact person for this update.
+      marketCode:
+        type: string
+        x-nullable: true
```
Comment (traskowskycaci) [inline swagger/support.yaml:1584]: Do we want this to be a nullable value?

File: pkg/services/address/address_creator.go (hunk 3 above)
Comment (danieljordan-caci) [reply pkg/services/address/address_creator.go:59]: Nah - it actually removes it if I try to put it there haha

File: swagger/support.yaml (hunk 4 above)
Comment (danieljordan-caci) [reply swagger/support.yaml:1584]: Yup! It's a nullable value in the db.

Comment (traskowskycaci) [reply swagger/support.yaml:1584]: Hmm.. since it will drive (probably) some important pricing data - do we need to it be not null?

Comment (traskowskycaci) [reply swagger/support.yaml:1584]: Discussed via slack - should make the db column itself not null. Resolving and will review again

**Comment by danieljordan-caci** [comment]
@traskowskycaci & @cameroncaci 

Updated relevant files per your requests - please take a look when ya get another chance!

**Review by traskowskycaci — APPROVED** [review]
Re-reviewed, still seeing the market code come through without issue. Looks good!

**Review by deandreJones — APPROVED** [review]
looks good
